typedb/stream/bidirectional_stream.py

```diff
@@ -78,8 +81,9 @@ def fetch(self, request_id: UUID) -> Union[transaction_proto.Transaction.Res, tr
                     raise TypeDBClientException.of(TRANSACTION_CLOSED)
                 server_msg = next(self._response_iterator)
             except RpcError as e:
-                self.close(e)
-                raise TypeDBClientException.of_rpc(e)
+                error = TypeDBClientException.of_rpc(e)
+                self.close(error)
+                raise error
```

Comment:
changed this so that we propagate our own exception instead of the gRPC error throughout the transaction queues